subsys/bluetooth/services/fast_pair/fp_crypto/include/fp_crypto_pn.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _FP_CRYPTO_PN_H_
+#define _FP_CRYPTO_PN_H_
+
+/**
+ * @defgroup fp_crypto_pn Fast Pair Personalized Name crypto
+ * @brief Internal API of Fast Pair Personalized Name crypto
+ *
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** Length of nonce (64 bits = 8 bytes). */
+#define FP_CRYPTO_PN_NONCE_LEN			8U
+/** Length of Additional Data packet header (128 bits = 16 bytes). */
+#define FP_CRYPTO_PN_ADDITIONAL_DATA_HEADER_LEN	16U
+
+/** Encrypt data using AES-CTR.
+ *
+ * @param[out] out Buffer to receive encrypted data.
+ * @param[in] in Plaintext data.
+ * @param[in] data_len Length of input and output data (in bytes).
+ * @param[in] key 128-bit (16-byte) AES key.
+ * @param[in] nonce 64-bit (8-byte) nonce.
+ *
+ * @return 0 If the operation was successful. Otherwise, a (negative) error code is returned.
+ */
+int fp_crypto_aes_ctr_encrypt(uint8_t *out, const uint8_t *in, size_t data_len, const uint8_t *key,
+			      const uint8_t *nonce);
+
+/** Decrypt data using AES-CTR.
+ *
+ * @param[out] out Buffer to receive plaintext data.
+ * @param[in] in Ciphertext data.
+ * @param[in] data_len Length of input and output data (in bytes).
+ * @param[in] key 128-bit (16-byte) AES key.
+ * @param[in] nonce 64-bit (8-byte) nonce used to encrypt data.
+ *
+ * @return 0 If the operation was successful. Otherwise, a (negative) error code is returned.
+ */
+int fp_crypto_aes_ctr_decrypt(uint8_t *out, const uint8_t *in, size_t data_len, const uint8_t *key,
+			      const uint8_t *nonce);
+
+/** Generate HMAC-SHA256.
+ *
+ * @param[out] out 256-bit (32-byte) buffer to receive hashed result.
+ * @param[in] in Input data.
+ * @param[in] data_len Length of input data.
+ * @param[in] aes_key 128-bit (16-byte) AES key used to encrypt data.
+ *
+ * @return 0 If the operation was successful. Otherwise, a (negative) error code is returned.
+ */
+int fp_crypto_hmac_sha256(uint8_t *out, const uint8_t *in, size_t data_len, const uint8_t *aes_key);
```

Comment:
I wouldn't implement it in backends as soon we will have to move to PSA API anyway (https://projecttools.nordicsemi.no/jira/browse/NCSDK-15251). Actually I'm planning to do it in next few weeks.

Reply:
IMO, PSA API should be implemented as another crypto backend because it does not have the necessary crypto operations for all Fast Pair extension combinations. If PSA API does not support HMAC SHA256, you can move the implementation from the fp_crypto_pn.c. By having a custom implementation using SHA256, you limit the performance for other backends that support this crypto operation.

Reply:
Discussed offline. Will be considered as a separate follow-up task (if we introduce a backend with native support of this crypto operations we could e.g. make these functions weak)

Reply:
https://projecttools.nordicsemi.no/jira/browse/NCSDK-16280 here is a task for it.